Dockerfile.ci

```diff
@@ -1338,25 +1250,10 @@ RUN echo "Airflow version: ${AIRFLOW_VERSION}"
 
 # Copy all scripts required for installation - changing any of those should lead to
 # rebuilding from here
-COPY --from=scripts install_packaging_tools.sh install_airflow_dependencies_from_branch_tip.sh \
-    common.sh /scripts/docker/
+COPY --from=scripts common.sh install_packaging_tools.sh install_additional_dependencies.sh /scripts/docker/
 
 # We are first creating a venv where all python packages and .so binaries needed by those are
 # installed.
-# In case of CI builds we want to pre-install main version of airflow dependencies so that
-# We do not have to always reinstall it from the scratch.
-# And is automatically reinstalled from the scratch every time patch release of python gets released
-# The Airflow and providers are uninstalled, only dependencies remain.
-# the cache is only used when "upgrade to newer dependencies" is not set to automatically
-# account for removed dependencies (we do not install them in the first place)
-#
-# We are installing from branch tip without fixing UV or PIP version - in order to avoid rebuilding the
-# base cache layer every time the UV or PIP version changes.
-RUN bash /scripts/docker/install_packaging_tools.sh; \
```

Comment:
We are removing installing from branch tip - combination of artifacts + remote + uv cache mounted from cache volume should bring all the necessary speedups in both - local and CI cases.